### PR TITLE
 streamingccl: tweak node lag replanning policy

### PR DESCRIPTION
**Pull request by msbutler**
This patch changes the policy for when pcr replans based on a lagging node:

- previously, lag node replanning would not occur if the replicated time had
  not advanced during the flow. This patch relaxes this requirement. If one
node is lagging behind all other nodes, and it is preventing the replicated
time from advancing, we really should replan!

- previously, lag node replanning would occur if the lagging node was more than
  x minutes behind the second most lagging node. But, if 2 nodes are lagging
behind all other nodes, we would not replan, even though we should. Instead,
this patch will trigger replanning if the lagging node's frontier is more than
x minutes behind the mean frontier of all participating nodes.

Overall, this patch relaxes the requirements for lag replanning, which will
become much less costly once #125044 lands, as it reduces the amount of wasted
work on each dist sql flow restart.

Informs #124915

Relase note: none

### CODE REVIEW

**Comment by msbutler** [comment]
unrelated flake

File: pkg/ccl/streamingccl/streamingest/node_lag_detector.go [hunk 1]
```diff
@@ -15,32 +15,37 @@ import (
...
 	ctx context.Context, executionDetails []frontierExecutionDetails, maxAllowableLag time.Duration,
 ) error {
 	if maxAllowableLag == 0 {
 		return nil
 	}
-	laggingNode, minLagDifference := computeMinLagDifference(executionDetails)
-	log.VEventf(ctx, 2, "computed min lag diff: %d lagging node, difference %.2f", laggingNode, minLagDifference.Minutes())
-	if maxAllowableLag < minLagDifference {
-		return errors.Wrapf(ErrNodeLagging, "node %d is %.2f minutes behind the next node. Try replanning", laggingNode, minLagDifference.Minutes())
+	laggingNode, maxLagDifference := computeMeanLagDifference(ctx, executionDetails)
```
Comment (dt) [inline pkg/ccl/streamingccl/streamingest/node_lag_detector.go:33]: max or mean?

File: pkg/jobs/jobspb/jobs.go [hunk 2]
```diff
@@ -44,3 +44,22 @@ func (fes RestoreFrontierEntries) Equal(fes2 RestoreFrontierEntries) bool {
 	}
 	return true
 }
+
+// RestoreFrontierEntries is a slice of RestoreFrontierEntries.
+// TODO(msbutler): use generics and combine with above.
+type ResolvedSpanEntries []ResolvedSpan
+
+func (fes ResolvedSpanEntries) Equal(fes2 ResolvedSpanEntries) bool {
```
Comment (dt) [inline pkg/jobs/jobspb/jobs.go:52]: "fes" ?

**Review by dt — APPROVED** [review]

File: pkg/jobs/jobspb/jobs.go (hunk 2 above)
Comment (msbutler) [reply pkg/jobs/jobspb/jobs.go]: lol, i think i blindly copied from the method defined above it. Fixed.

**Comment by msbutler** [comment]
TFTR!

bors r=dt

**Pull request merged**